specs/rollup-node.md

```diff
@@ -175,58 +235,77 @@ The following scenarios are assimilated to errors:
-  - Find the next L1 block (it may not exist yet) and return that as `nextRefL1`, along with the `refL2`.
-    - Note: after looking up `N+1` ensure L1 has not changed during block-by-number lookups (`refL1 == nextL1_parent`).
-- While have not found a block in the engine common with the canonical chain, traverse the L2 chain back until genesis:
-  - Each step starts by caching the previous `currentL1` as `nextRefL1`.
-  - Lookup the parent by hash: Each step `refL2` should equal the previous `parentL2`.
-    `refL1` and `parentL2` are also traversed back by parsing the `refL2` block.
-  - The canonical L1 block is looked up at the same height (`currentL1`).
-    - If not found: consider this a reorg to a shorter L1 chain, continue.
-  - If the engine and canonical chain match (`refL1 == currentL1`), then return that as `nextRefL1`, along with `refL2`.
-- If there are no common blocks after genesis, check if `refL2` and `currentL1` match the expected genesis blocks.
-- If the genesis is correct, the last cached `nextRefL1` is returned, along with the L2 genesis.
-
-> Note that post-[merge], the L1 chain will offer finalization guarantees meaning that it won't be able to re-org more
-> than `FINALIZATION_DELAY_BLOCKS == 50400` in the past, hence preserving our finalization guarantees.
+[l1-finality]: https://www.paradigm.xyz/2021/07/ethereum-reorgs-after-the-merge/
 
-[merge]: https://ethereum.org/en/eth2/merge/
+> **TODO** the spec doesn't encode the notion of fault proof yet, revisit this (and include links) when it does
 
-Just like before, the meaning of errors returned by RPC calls is unspecified and must be handled at the implementer's
-discretion, while remaining compatible with the specification.
+## Whole L2 Chain Derivation
 
-## Finalization Guarantees
+The previous two sections presents an inductive process: given that we know the "current" L2 block, well as the next L1
+block, then we can derive [payload attributes] for the next L1 block, and from that the next L2 block.
 
-[finalization]: #finalization-guarantees
+To derive the whole L2 chain from scratch, we simply start with the L2 genesis block as the current L2 block, and the
+block at height `L2_CHAIN_INCEPTION + 1` as the next L1 block. Then we iteratively apply the derivation process from the
+previous section to each successive L1 block until we have caught up with the L1 head.
 
-As already alluded to in the section on [interacting with the execution engine][calling-exec-engine], an L2 block is
-considered *finalized* after a delay of `FINALIZATION_DELAY_BLOCKS == 50400` blocks after the L1 block that generated
-it. This is a duration of approximately 7 days worth of L1 blocks.
+> **TODO** specify genesis block
 
-L1 Ethereum [reaches finality approximately every 12 minutes][l1-finality], so these L2 blocks can safely be considered
-to be final: they will never disappear from the chain's history because of a re-org.
+# Handling L1 Re-Orgs
 
```

Comment:
I want to move this to a implementation notes directory, cf. https://github.com/ethereum-optimism/optimistic-specs/issues/141

Any objections?